pymeasure/instruments/keysight/keysightN7776C.py

```diff
@@ -0,0 +1,256 @@
+
+    sweep_step = Instrument.control('sour0:wav:swe:step?','sour0:wav:swe:step %fnm',
+                                    """ Step width of th[e sweep (in nanometers).""",
+                                    validator=strict_range,
+                                    values=[0.0001,WL_RANGE[1]-WL_RANGE[0]])
+    sweep_speed = Instrument.control('sour0:wav:swe:speed?','sour0:wav:swe:speed %fnm/s',
+                                    """ Speed of the sweep (in nanometers per second).""",
+                                    validator=strict_discrete_set,
+                                    values=[0.5,1,50,80,200],
+                                    get_process=lambda v: v*1e9)
+    sweep_mode = Instrument.control('sour0:wav:swe:mode?','sour0:wav:swe:mode %s',
+                                    """ Sweep mode of the swept laser source """,
+                                    validator=strict_discrete_set,
+                                    values=['STEP','MAN','CONT'])
+    sweep_twoway = Instrument.control('sour0:wav:swe:rep?','sour0:wav:swe:rep %s',
+                                    """Sets the repeat mode. Applies in stepped,continuous and manual sweep mode.""",
+                                    validator=strict_discrete_set,
+                                    values=[True,False],
+                                    set_process=lambda v: ['ONEW','TWOW'][int(v)],
+                                    get_process=lambda v: bool(['ONEW','TWOW'].index(v)))
+    _sweep_check = Instrument.measurement('sour0:wav:swe:chec?',
+                                    """Returns whether the currently set sweep parameters (sweep mode, sweep start, stop, width, etc.) are consistent. If there is a
+                                    sweep configuration problem, the laser source is not able to pass a wavelength sweep.""")
+
+    sweep_points = Instrument.measurement('sour0:read:points? llog',
+                                    """Returns the number of datapoints that the :READout:DATA? command will return.""")
+    
+    sweep = Instrument.control('sour0:wav:swe?','sour0:wav:swe %g',
+                                    """ State of the wavelength sweep. Stops, starts, pauses 
+                                    or continues a wavelength sweep.""")
+
+    wl_logging = Instrument.control('SOUR0:WAV:SWE:LLOG?','SOUR0:WAV:SWE:LLOG %g',
+                                    """ State (on/off) of the lambda logging feature of the laser source.""",
+                                    validator=strict_discrete_set,
+                                    values=[True,False],
+                                    set_process=lambda v: int(v),
+                                    get_process=lambda v: bool(v))
+
+    def check_sweep_params(self):
+        response = self._sweep_check
+        if response[0] == 0.0:
+            return True
+        elif response[0] == 368:
+            log.log(30,'Warning: End Wavelength <= Start Wavelength.')
```

Comment:
please instead
```suggestion
            log.warning('End Wavelength <= Start Wavelength.')
```
etc.